test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13019,21 +13019,17 @@ def __init__(self, username=None, attrs={}, jobname=None):
         if self.platform == 'cray' or self.platform == 'craysim':
             if 'Resource_List.select' in attrs:
                 ra = []
-                select = attrs['Resource_List.select']
-                c = select.startswith('vntype=') or ':vntype=' in select
-                m = '+vntype=' in select or c
-                r = select.split('+')
-                if not m:
+                select = PbsTypeSelect(attrs['Resource_List.select'])
+                if 'vntype' not in select.resources.keys():
```

Comment:
Isn't it still wrong? if one of the chunk requested vntype and others didn't then it will still show vntype be part of select.resources dict right?

Reply:
@arungrover , yes you are right it will still show vntype in select.resources dict but as per design doc  if vntype not present in select statement (means any of the chunk) then only we will be adding vntype to each chunk. If any of the chunk contains vntype then we will not modify select specification and will use the same select as user has provided.
Example:
case-1 select without vntype:
select=1:ncpus=1+1:ncpus=2
This will get modified to:
select=1:ncpus=1:vntype=cray_compute+1:ncpus=2:vntype=cray_compute

case-2 select with one of the chunk containing vntype
select=1:ncpus=1+1:ncpus=2:vntype=cray_compute
In this case it will use the same select statement as it is without modifying it.

I have also attached the execution logs to tickets. Let me know if its still confusing.

Reply:
@borlesanket I think this is not correct. Can you point me to the design where we listed this caveat. It still seems wrong to me, we should add vntype to all weighted chunks that are listed in the select specification (if vntype is not present).

Reply:
@arungrover , you are right we should add vntype to all chunks if vntype is not present in select statement and this is what I was trying to say in the case-1 above.
But if any of the chunk contains vntype then we are not altering the select statement and it will remain as it is.
In design doc if you see inside Job.__init__() inside that I am saying that:
If select statement is present in attrs:
- If it does not contain vntype then assign vntype value as 'cray_compute' to each select chunk
- If vntype present in select statement then do not modify it.

I am just clearing on case-2:
Are you trying to say that in case-2 also we should add the vntype to all chunks apart from from one chunk which already had vntype already?

Reply:
This looks like a design discussion, so its better to discuss on forum.
http://community.pbspro.org/t/pp-946-on-cray-a-ptl-script-setting-select-specification-on-a-job-using-set-attributes-method-makes-the-job-fail-to-run/674